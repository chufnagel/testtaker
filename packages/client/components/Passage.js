import PropTypes from "prop-types";

const Passage = ({ passage }) => (
  <div>
    {passage.map(entry => (
      <p className="passage" key={entry.line}>
        {`${entry.line}: ${entry.text}`};
      </p>
    ))}
  </div>
);

Passage.propTypes = {
  passage: PropTypes.array,
};

Passage.defaultProps = {
  passage: [
    {
      line: "1",
      text: "Ergo iter inceptum peragunt fluvioque propinquant.",
    },
    {
      line: "2",
      text: "Navita quos iam inde ut Stygia prospexit ab unda",
    },
  ],
};

export default Passage;
