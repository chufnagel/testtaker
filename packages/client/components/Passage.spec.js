/* global beforeEach describe, expect, test */
import { shallow } from "enzyme";
import Passage from "./Passage";

// Passage.defaultProps = {
//   passage: [
//     {
//       line: "1",
//       text: "Ergo iter inceptum peragunt fluvioque propinquant.",
//     },
//     {
//       line: "2",
//       text: "Navita quos iam inde ut Stygia prospexit ab unda",
//     },
//   ],
// };

describe("Passage", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Passage />);
  });
  test("It should render without crashing", () => {
    expect(wrapper.exists()).toEqual(true);
  });
  test("Passage matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
