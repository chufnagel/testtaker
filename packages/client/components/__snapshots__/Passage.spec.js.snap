// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passage Passage matches snapshot 1`] = `
<div>
  <p
    className="passage"
    key="1"
  >
    1: Ergo iter inceptum peragunt fluvioque propinquant.
    ;
  </p>
  <p
    className="passage"
    key="2"
  >
    2: Navita quos iam inde ut Stygia prospexit ab unda
    ;
  </p>
</div>
`;
